worker/worker.go

```diff
@@ -173,18 +165,45 @@ func detectFeatures(name string, data map[string][]byte, namespace *database.Nam
 		return
 	}
 
-	// Ensure that every feature has a Namespace associated, otherwise associate the detected
-	// namespace. If there is no detected namespace, we'll throw an error.
+	// If there are no FeatureVersion detected, use parent's FeatureVersions if possible.
+	// TODO(Quentin-M): We eventually want to give the choice to each detectors to use none/some
+	// parent's FeatureVersions. It would be useful for detectors that can't find their entire result
+	// using one Layer.
+	if len(features) == 0 && parent != nil {
+		features = parent.Features
+		return
+	}
+
+	// Ensure that each FeatureVersion has a Namespace associated.
+	// If no Namespace is associated,
+	// - If the FeatureVersion is present in the parent layer, associate its Namespace,
+	// - If a Namespace has been detected in the layer, associate it,
+	// - Otherwise, throw an error.
```

Comment:
Let's add a comment addressing the building of this map and then we'll add comments inside the next block to describe each of these cases.

```golang
// Build a map of the namespaces for each FeatureVersion in our parent layer.
```